src/ripple/rpc/handlers/AMMInfo.cpp

```diff
@@ -155,7 +155,8 @@ doAMMInfo(RPC::JsonContext& context)
                 auctionSlot);
             auction[jss::time_interval] = timeSlot ? *timeSlot : 0;
             auctionSlot[sfPrice].setJson(auction[jss::price]);
-            auction[jss::discounted_fee] = auctionSlot[sfDiscountedFee];
+            auction[jss::discounted_fee] =
+                auctionSlot[~sfDiscountedFee].value_or(0);
```

Comment:
This can safely be changed back to the original code `auction[jss::discounted_fee] = auctionSlot[sfDiscountedFee];`. If the field isn't set, it'll return 0.